src/herald/featureherald.js

```diff
@@ -94,7 +96,14 @@ olgm.herald.Feature.prototype.activate = function() {
 
   // event listeners (todo)
   var keys = this.listenerKeys;
-  keys.push(geometry.on('change', this.handleGeometryChange_, this));
+  this.geometryChangeKey_ = geometry.on('change',
+                                        this.handleGeometryChange_,
+                                        this);
```

Comment:
on 2 lines:

this.geometryChangeKey_ = geometry.on(
    'change', this.handleGeometryChange_, this);